### PR TITLE
ICDS Dashboard: Adding new columns to SDD at sector level

### PR DESCRIPTION
**Pull request by saiprakash-v**
Asana ticket: https://app.asana.com/0/1158698071595857/1158299139323610
##### SUMMARY

Added number of cbe and vhsnd conducted columns to SDD.
Removed the Doc from here as this was open to all -: Rohit Negi
##### FEATURE FLAG

Newly added columns are under feature flag
SDD is already under feature flag in mobile dashboard

### CODE REVIEW

**Comment by czue** [comment]
I don't have much context on this so will let @Rohit25negi or @calellowitz review

**Review by czue — APPROVED** [review]
Sorry, meant code looks good but maybe someone else can sign off who knows the requirements.

**Comment by saiprakash-v** [comment]
All the tests passed locally. Not sure why build is failing here.

**Comment by czue** [comment]
build fail did not look related. I restarted it. @Rohit25negi or @calellowitz would you be able to take a second look at this from a requirements perspective?

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js [hunk 1]
```diff
@@ -197,6 +198,25 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
         ],
     };
 
+    if (haveAccessToFeatures) {
+        var numberOfCbeConducted = {
+            'mData': 'num_awcs_conducted_cbe',
+            'heading': 'Number of CBE Conducted',
+            'tooltipValue': 'Number of Community-Based Events (CBEs) organised by an AWC in a month',
+            'columnValueType': 'raw',
+            'columnValueIndicator': 'num_awcs_conducted_cbe',
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:207]: Hey, if you want the number of CBE/VHND itself, the columns to use are `cbe_conducted, vhnd_conducted,`.   `num_awcs_conducted_cbe` is the number of AWCS conducted at least 2 CBE. and `num_awcs_conducted_vhnd` is number of awcs conducted at least 1 CBE.

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Left one comment. I think the columns used are not right.

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js (hunk 1 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:207]: I agree. But the definition that you mentioned is for all the levels above sector right? In sector level, I think `num_awcs_conducted_cbe` returns cbe conducted by individual awc in the sector.

Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:207]: These two columns are being added only at the sector level

Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:207]: which tooltip you refered? I just checked it says `if awcs has conducted at least 2 Cbes then yes else no`.
These columns are the actual count of CBE and VHND yeah?
if you see `num_awcs_conducted_cbe` is calculated here as if the CBE count is>1 then its counted as 1 [here](https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py#L100). The actual count is stored [here](https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py#L101)

**Comment by Rohit25negi** [comment]
@saiprakash-v  is this ready for next round of review
?

**Comment by saiprakash-v** [comment]
> @saiprakash-v is this ready for next round of review
> ?

Yes @Rohit25negi

**Comment by saiprakash-v** [comment]
@Rohit25negi Gentle reminder. This pr is ready for review

File: custom/icds_reports/migrations/0186_refresh_views.py [hunk 2]
```diff
@@ -12,4 +12,4 @@ class Migration(migrations.Migration):
         ('icds_reports', '0179_location_deprecation_columns'),
```
Comment (Rohit25negi) [inline custom/icds_reports/migrations/0186_refresh_views.py:12]: What are these changes?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
All looks good except you will need to fix the migration.

File: custom/icds_reports/migrations/0185_update_sdd_view.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (saiprakash-v) [reply custom/icds_reports/migrations/0185_update_sdd_view.py:1]: 0185 isn't in the migration list .0186 and 187 are generated after this. This is uptodate with the master

File: custom/icds_reports/migrations/0186_refresh_views.py (hunk 2 above)
Comment (saiprakash-v) [reply custom/icds_reports/migrations/0186_refresh_views.py:12]: I guess, these are @calellowitz migration files. Only 185 is related to this PR

**Comment by saiprakash-v** [comment]
> All looks good except you will need to fix the migration.

@Rohit25negi If this PR is good to go, can you approve it? I will update the migration files in https://github.com/dimagi/commcare-hq/pull/27379

**Review by Rohit25negi — APPROVED** [review]
pr looks good. could you update the migration please to follow the sequence? 
Another thing is, we should not be adding the google docs to the PRs as these docs may be open to all as it was in this case. I know we not doing a great job with maintaining the permission of google docs. But we should only add the ticket link in the description.

**Comment by saiprakash-v** [comment]
> pr looks good. could you update the migration please to follow the sequence?

Updated it.
> Another thing is, we should not be adding the google docs to the PRs as these docs may be open to all as it was in this case. I know we not doing a great job with maintaining the permission of google docs. But we should only add the ticket link in the description.

Sure.

**Comment by Rohit25negi** [comment]
@saiprakash-v  I think the changes are good in this PR. It just needs to be fixed for migration conflict.
NOTE: I think I also saw one of your Pr have same conflict. So Better would be to fix one and let it merge and then fix the other so that you don't need to re-fix one of them.

**Comment by saiprakash-v** [comment]
> @saiprakash-v I think the changes are good in this PR. It just needs to be fixed for migration conflict.
> NOTE: I think I also saw one of your Pr have same conflict. So Better would be to fix one and let it merge and then fix the other so that you don't need to re-fix one of them.

Ok. I updated migration files in LS launched pr. Will update this after that

**Comment by Rohit25negi** [comment]
@saiprakash-v  I merged you LS pr, you can fix the migration in this.

**Comment by Rohit25negi** [comment]
good to go once the build passes

**Pull request merged**